graph/patterns/navigation-property.md

```diff
@@ -0,0 +1,185 @@
+# Navigation Property
+
+Microsoft Graph API Design Pattern
+
+*A navigation property is used to identify a relationship between resources.*
+
+## Problem
+--------
+
+It is often valuable to represent a relationship between resources in an API. This may be a many-to-one or a one-to-many relationship. 
+
+Relationships between resources are often implicitly represented by a property contained in one of the resources that identifies the other related resource. Usually that information returned in a representation as an id value and the property is named using a convention that identifies the target type of related resource. e.g. userId 
+
+In many-to-one relationships, for a client to access the related resource it must request the primary resource, read the id value of the related resource and then construct a URL to the related resource using the Id value and knowledge of how the property name maps to the related resource. This requires at least two round trips and requires the client know how to construct the URL to the related resource.
+
+For both many-to-one and one-to-many relationships in order to retrieve information from both resources on both sides of the relationship requires at minimum two requests.
```

Comment:
We seem to be repeating the same info here.  Can this merge into line 14 with "For relationships, for a client..."
Or at worst "For both many-to-one and one-to-many relationships, for a client...